### ISSUE TITLE
[Feature]: Set proxy mode true/false

### ISSUE THREAD

**Issue by ablax** [citizenfx/txAdmin#535]
### Scope

Web

### Feature Description

Set proxy mode true/false

### Use Case

If someone want's to put txadmin behind proxy (cloudflare for example) they should still be able to get admins ip.

### Proposed Solution

Add Convar

**Comment by Comet1903** [comment]
This is not possible, this is also how Tor works... Basically you can't get the admins IP... You can reverse proxy txAdmin on your own server, so you can view the admins IP in your proxy's log. (on my server I use nginx to do exactly what you want to do)
If that what I am saying here is not true then please correct me

**Comment by ablax** [comment]
We are not talking about Tor here. When you run txadmin on your VPS, and an admin logs in, it will show the admin's IP in txadmin log. But having in mind that I do use nginx reverse proxy (just like you said) the txadmn no longer receives admin's IP, but instead it displays 127.0.0.1 (loopback address) because of the local proxy, and if you set it to add an header for forwarded-for address (just like cloudflare adds as well) and you set app.proxy to true koa automatically knows to use the header instead the IP making the request.

**Comment by Rosw3lluk** [comment]
Im pretty certain if you use cloudflare, or i hhave in the past, the authentic origin ip is stored  as a header sent by cloudflare, which is the origin IP of the person making the request, and wont tell you cloudflares IP, PHPBB use this for there forum admin logs.

**Comment by ablax** [comment]
Yes the IP is stored as a header from the proxy providers. And the purpose of this is to be able to get that IP, otherwise, txAdmin would be using Cloudflare's IP.

**Comment by Doxylamin** [comment]
txAdmin should/could just use X-RealIP / X-Forwarded-For, if either of those are set.

**Comment by ablax** [comment]
> txAdmin should/could just use X-RealIP / X-Forwarded-For, if either of those are set.

Pretty much it would be the same as enabling proxy mode. There is no difference.

**Comment by Doxylamin** [comment]
Except there would be no need for such proxy mode and/or manual changes, as it would be detected automatically.

**Comment by ablax** [comment]
If you enable proxy mode (you can just do it in the code without var, it's disabled on purpose) it will also detect it automatically and will work as you described it. That's the whole purpose of it
